x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/execution/PlanExecutor.java

```diff
@@ -83,13 +86,16 @@ public void esql(
         QueryMetric clientId = QueryMetric.fromString("rest");
         metrics.total(clientId);
 
+        var begin = System.nanoTime();
         ActionListener<Result> executeListener = wrap(x -> {
             planTelemetryManager.publish(planTelemetry, true);
+            slowLog.onQueryPhase(x, request.query());
```

Comment:
To help with encapsulation, please create two internal methods called when the query is successfully executed or is failing and move the metrics, telemetry and now the slow log there (e.g. onQuerySuccesful(), onQueryFailure())